cve_bin_tool/checkers/libsolv.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+CVE checker for libsolv
+
+https://www.cvedetails.com/vulnerability-list/vendor_id-8184/product_id-51703/Opensuse-Libsolv.html
+
+"""
+
+from cve_bin_tool.checkers import Checker
+
+
+class LibsolvChecker(Checker):
+    CONTAINS_PATTERNS = []
+    FILENAME_PATTERNS = [r"libsolv.so", r"libsolvext.so"]
+    VERSION_PATTERNS = [
+        r"libsolv-([0-9]+\.[0-9]+\.[0-9]+)",
+        r"libsolv.so.1-([0-9]+\.[0-9]+\.[0-9]+)",
+    ]
```

Comment:
```suggestion
    VERSION_PATTERNS = [r"libsolv(?:\.so\.1)?-([0-9]+\.[0-9]+\.[0-9]+)"]
```

You can use the pattern above and can decrease the number of scans on a package (even though the performance increment is minimal). 

What happens here is that `.so.1` is considered optional using `?` at the end of the brackets and is also denoted as a non-capturing group using `?:` inside the brackets. So the only group detected would be the version strings.